roles/coredns/tasks/main.yml

```diff
@@ -0,0 +1,13 @@
+- name: Set node selector for CoreDNS components
+  kubernetes.core.k8s:
+    definition:
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: coredns
+        namespace: kube-system
+      spec:
+        template:
+          spec:
+            nodeSelector: "{{ coredns_node_selector }}"
+    wait: true
```

Comment:
coredns is installed by kubeadm so i suggest that we can move this patch to `kubernetes` role

Reply:
i think control-plane.yml file is reasonable? Also please set `run_once: true` in the task.